src/integrations/GoogleCloudCommonIntegration.ts

```diff
@@ -0,0 +1,215 @@
+                            as no tracer is available`);
+                        return original.apply(this, [options, config, callback]);
+                    }
+
+                    const originalFunction = integration.getOriginalFunction(wrappedFunctionName);
+
+                    const parentSpan = tracer.getActiveSpan();
+
+                    const uri = options.uri.split(URL_SPLIT_STRING)[1] || '';
+                    const splitUri = uri.split('/');
+                    const service = splitUri[0] || 'google-cloud';
+                    const projectId = splitUri[3] || '';
+                    const path = uri.split(`${projectId}/`)[1];
+                    const operation = path.split('/')[0] || '';
+                    const queryStatement = GoogleCloudCommonIntegration.getQueryStatement(options);
+
+                    ThundraLogger.debug(`<GoogleCloudCommonIntegration> Starting makeRequest span with name ${service}`);
+
+                    span = tracer._startSpan(service, {
+                        childOf: parentSpan,
+                        domainName: DomainNames.API,
+                        className: ClassNames.GOOGLE_BIGQUERY,
+                        disableActiveStart: true,
+                    });
+
+                    const tags = {
+                        [GoogleCommonTags.PROJECT_ID]: projectId,
+                        [GoogleCommonTags.SERVICE]: service,
+                        [GoogleBigQueryTags.OPERATION]: operation,
+                        ...(queryStatement ? { [GoogleBigQueryTags.QUERY]: queryStatement } : undefined),
+                        [SpanTags.OPERATION_TYPE]: GoogleCommonOperationTypes.QUERY,
+                        [SpanTags.SPAN_TYPE]: SpanTypes.GOOGLE_BIGQUERY,
+                        [SpanTags.TOPOLOGY_VERTEX]: true,
+                    };
+
+                    span.addTags(tags);
+                    const wrappedCallback = (err: any, res: any)  => {
+                        if (!span) {
+                            if (originalCallback) {
+                                originalCallback(err, res);
+                            }
+
+                            return;
+                        }
+
+                        if (err) {
+                            span.setErrorTag(err);
+                        } else if (res) {
+                            const responseSize = sizeof(res);
+                            const responseMaxSize = ConfigProvider.get<number>(
+                                ConfigNames.THUNDRA_TRACE_INTEGRATIONS_GOOGLE_BIGQUERY_RESPONSE_SIZE_MAX);
+                            if (responseSize <= responseMaxSize) {
+                                span.addTags({
+                                    [GoogleBigQueryTags.RESPONSE]: res,
+                                });
+                            } else if (res.jobReference && res.jobReference.jobId) {
+                                span.addTags({
```

Comment:
Why we add this tag in else case (if response size is bigger than limit)?

Reply:
Can we use HTTP method name (`GET`, `PUT`, `POST`, `PATCH`, `DELETE`) to decide operation type?
For example:
- `GET` => `READ`
- `PUT`, `POST`, `PATCH` => `WRITE`
- `DELETE` -> `DELETE`